README.md

```diff
@@ -1,2 +1,47 @@
 # secret-common-lib
-secret common lib to handle secret created on managed or un-managed k8s cluster on IBM Cloud
+secret common lib library handles authentication for access to ibmcloud resources by implementing secret providers.
+
+## secret providers
+
+The following interface can be seen defined in the secret-utils-lib library
+```
+// SecretProviderInterface ...
+type SecretProviderInterface interface {
+
+	// GetIAMToken ...
+	GetIAMToken(secret string, freshTokenRequired bool) (string, uint64, error)
+
+	// GetDefaultIAMToken ...
+	GetDefaultIAMToken(freshTokenRequired bool) (string, uint64, error)
+}
+```
+
+## Pre requisites
+- An environment variable IKS_ENABLED must to be set to true or false.
+- A k8s secret must be present in the same namespace where the pod (the application in which this code is used) is deployed.
+- The secrets format are present in secrets folder of secret-utils-lib library - ibm-cloud-credentials.yaml or storage-secret-store.yaml
```

Comment:
This point is bit confusing, can you add a sample secret format or more details.

Reply:
I have added the secret format in this repo